kafka/main.go

```diff
@@ -67,92 +88,145 @@ func StartConsumers(providedServer *server.Server, logger *zerolog.Logger) error
 		topics = append(topics, topicMapping.Topic)
 	}
 
-	consumerCount, err := strconv.Atoi(os.Getenv("KAFKA_CONSUMERS_PER_NODE"))
+	reader := newConsumer(topics, adsConsumerGroupV1, logger)
+
+	batchPipeline := make(chan *MessageContext)
+	ctx := context.Background()
+	go processMessagesIntoBatchPipeline(ctx, topicMappings, providedServer, reader, batchPipeline, logger)
+	for {
+		readAndCommitBatchPipelineResults(ctx, reader, batchPipeline, logger)
+	}
+}
+
+// readAndCommitBatchPipelineResults does a blocking read of the batchPipeline channel and
+// then does a blocking read of the errorResult in the MessageContext in the batchPipeline.
+// When an error appears it means that the message processing has entered a finalized state
+// and is either ready to be committed or has encountered a temporary error. In the case
+// of a temporary error, the application panics without committing so that the next reader
+// gets the same message to try again.
+func readAndCommitBatchPipelineResults(
+	ctx context.Context,
+	reader *kafka.Reader,
+	batchPipeline chan *MessageContext,
+	logger *zerolog.Logger,
+) {
+	msgCtx, ok := <-batchPipeline
+	if !ok {
+		logger.Error().Msg("batchPipeline channel closed")
+		panic("batch item error")
```

Comment:
were we going to intentionally close the `batchPipeline` channel for these panic cases? I guess since this is only ever called in the main goroutine it doesn't matter much since this will kill the entire program?

Reply:
Yes that was my thinking. No need to close in this panic case as far as I know.

Reply:
I think when I traced things out this wasn't actually running in the main thread. i'm worried that if we panic here things won't properly recover as a result

Reply:
fixed as discussed